src/portgraph.rs

```diff
@@ -1584,6 +1627,77 @@ impl<'a> DoubleEndedIterator for Neighbours<'a> {
 
 impl<'a> FusedIterator for Neighbours<'a> {}
 
+/// Iterator over the links connecting two nodes, created by
+/// [`PortGraph::get_connections`].
+#[derive(Clone)]
+pub struct NodeConnections<'a> {
+    graph: &'a PortGraph,
+    target: NodeIndex,
+    port_links: Zip<NodePorts, NodeLinks<'a>>,
+}
+
+impl<'a> NodeConnections<'a> {
+    /// Create a new iterator over the links connecting two nodes, from an
+    /// iterator over the ports and links.
+    pub fn new(
+        graph: &'a PortGraph,
+        target: NodeIndex,
+        ports: NodePorts,
+        links: NodeLinks<'a>,
+    ) -> Self {
+        Self {
+            graph,
+            target,
+            port_links: ports.zip(links),
+        }
+    }
+
+    /// Checks if we can yield an element from the `port_links` iterator.
+    ///
+    /// Receives `self.graph` and `self.target` as arguments to avoid borrowing issues.
+    fn get_next(
+        graph: &PortGraph,
+        target: NodeIndex,
+        next: (PortIndex, Option<PortIndex>),
+    ) -> Option<(PortIndex, PortIndex)> {
+        let (from, to) = next;
+        let to = to?;
+        if graph.port_node(to) == Some(target) {
+            Some((from, to))
+        } else {
+            None
+        }
+    }
+}
+
+impl<'a> Iterator for NodeConnections<'a> {
+    type Item = (PortIndex, PortIndex);
+
+    #[inline]
+    fn next(&mut self) -> Option<Self::Item> {
+        self.port_links
+            .find_map(|next| Self::get_next(self.graph, self.target, next))
+    }
+
+    #[inline]
+    fn size_hint(&self) -> (usize, Option<usize>) {
+        self.port_links.size_hint()
+    }
+}
+
+impl<'a> DoubleEndedIterator for NodeConnections<'a> {
+    #[inline]
+    fn next_back(&mut self) -> Option<Self::Item> {
+        self.port_links
```

Comment:
not sure why this doesn't just use port_links.next_back, might be worth a comment